graphql/resolvers.go

```diff
@@ -632,6 +634,26 @@ func (r *mutationResolver) SetTaskPriority(ctx context.Context, taskID string, p
 	return &apiTask, nil
 }
 
+func (r *mutationResolver) SchedulePatch(ctx context.Context, patchID string, reconfigure PatchReconfigure) (*restModel.APIPatch, error) {
+	patchUpdateReq := PatchVariantsTasksRequest{}
+	patchUpdateReq.BuildFromGqlInput(reconfigure)
+	version, err := r.sc.FindVersionById(patchID)
+	// FindVersionById does not distinguish between nil version err and db err; therefore must check that err
+	// does not contain nil version err values before sending InternalServerError
+	if !strings.Contains(err.Error(), strconv.Itoa(http.StatusNotFound)) {
```

Comment:
This will panic if err is nil. You should put this in an if block, and also return an error if the patch isn't found

Reply:
no error should be returned if the patch is not found because the patch may have not been scheduled yet; therefore there is not one in versions collection.

Reply:
You should still put this in an if block because calling `.Error()` on something that is nil will panic.

Reply:
i put it in an if statement. however before the if block, `.Error()` was being called on a `nil` error without a panic ocurring.

Reply:
Are you sure? Is it possible it wasn't being called somehow? I'm not sure what code you're referring to, but for example, https://play.golang.org/p/KkRy-1X1FN7

Reply:
it's the line of code this is commenting on